devicetypes/smartthings/fibaro-heat-controller.src/fibaro-heat-controller.groovy

```diff
@@ -0,0 +1,334 @@
+			break
+		case "off":
+			modeValue = 0
+			break
+	}
+
+	[
+			secureEncap(zwave.thermostatModeV2.thermostatModeSet(mode: modeValue), 1),
+			"delay 2000",
+			secureEncap(zwave.thermostatModeV2.thermostatModeGet(), 1)
+	]
+}
+
+def heat() {
+	setThermostatMode("heat")
+}
+
+def off() {
+	setThermostatMode("off")
+}
+
+def emergencyHeat() {
+	setThermostatMode("emergency heat")
+}
+
+def setHeatingSetpoint(setpoint) {
+	setpoint = temperatureScale == 'C' ? setpoint : fahrenheitToCelsius(setpoint)
+	[
+			secureEncap(zwave.thermostatSetpointV2.thermostatSetpointSet([precision: 1, scale: 0, scaledValue: setpoint, setpointType: 1, size: 2]), 1),
+			"delay 2000",
+			secureEncap(zwave.thermostatSetpointV2.thermostatSetpointGet(setpointType: 1), 1)
+	]
+}
+
+def setThermostatSetpointUp() {
+	def setpoint = device.latestValue("heatingSetpoint")
+	if (setpoint < maxHeatingSetpointTemperature) {
+		setpoint = setpoint + (temperatureScale == 'C' ? 0.5 : 1)
+	}
+	setHeatingSetpoint(setpoint)
+}
+
+def setThermostatSetpointDown() {
+	def setpoint = device.latestValue("heatingSetpoint")
+	if (setpoint > minHeatingSetpointTemperature) {
+		setpoint = setpoint - (temperatureScale == 'C' ? 0.5 : 1)
+	}
+	setHeatingSetpoint(setpoint)
+}
+
+def refresh() {
+	def cmds = [
+			secureEncap(zwave.batteryV1.batteryGet(), 1),
+			secureEncap(zwave.batteryV1.batteryGet(), 2),
+			secureEncap(zwave.thermostatSetpointV2.thermostatSetpointGet(setpointType: 1)),
+			secureEncap(zwave.thermostatModeV2.thermostatModeGet()),
+			secureEncap(zwave.sensorMultilevelV5.sensorMultilevelGet()),
+			secureEncap(zwave.configurationV1.configurationGet(parameterNumber: 3))
+	]
+
+	delayBetween(cmds, 2500)
+}
+
+def ping() {
+	refresh()
+}
+
+private secureEncap(cmd, endpoint = null) {
+	secure(encap(cmd, endpoint))
+}
+
+private secure(cmd) {
+	if(zwaveInfo.zw.endsWith("s")) {
+		zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+	} else {
+		cmd.format()
+	}
+}
+
+private encap(cmd, endpoint = null) {
+	if (endpoint) {
+		zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint:endpoint).encapsulate(cmd)
+	} else {
+		cmd
+	}
+}
+
+def switchMode() {
+	def currentMode = device.currentValue("thermostatMode")
+	def supportedModes = state.supportedModes
+	if (supportedModes && supportedModes.size()) {
+		def next = { supportedModes[supportedModes.indexOf(it) + 1] ?: supportedModes[0] }
+		def nextMode = next(currentMode)
+		setThermostatMode(nextMode)
+	} else {
+		log.warn "supportedModes not defined"
+	}
+}
+
+def sendEventToChild(event) {
+	String childDni = "${device.deviceNetworkId}:2"
+	def child = childDevices.find { it.deviceNetworkId == childDni }
+	child?.sendEvent(event)
+}
+
+private refreshChild() {
+	def cmds = [
+			secureEncap(zwave.batteryV1.batteryGet(), 2),
+			secureEncap(zwave.sensorMultilevelV5.sensorMultilevelGet(), 2)
+	]
+	sendHubCommand(cmds, 2000)
+}
+
+private forcedRefresh() {
+	sendHubCommand(refresh())
+}
+
+def addChild() {
```

Comment:
I believe we came to the conclusion that we can only commit to supporting this device if it has the temperature sensor installed. We should create this device automatically on add.

Reply:
On second thought, this is the correct approach, but the cert team should be aware that we're only certifying the device if it's installed with the temp. sensor.